bigquery_etl/cli/stage.py

```diff
@@ -139,6 +139,9 @@ def deploy(
         name = artifact_file.parent.name
         test_path = TEST_DIR / project / dataset / name
 
+        if "INFORMATION_SCHEMA" in name:
+            continue
+
```

Comment:
We don't want to create any query or schema files for `INFORMATION_SCHEMA`

Reply:
Yeah, I guess creating this file is necessary to have references to `some-project.region-us.INFORMATION_SCHEMA.JOBS_BY_PROJECT` updated to `bigquery-etl-integration-test.region-us.INFORMATION_SCHEMA.JOBS_BY_PROJECT`. Not copying them ensures that we don't try to deploy them to the stage environment.

Reply:
As it is, if the `INFORMATION_SCHEMA` references are updated then it'll be to something commit-hash-specific in `bigquery-etl-integration-test`, which might work for `{project}.{dataset}.INFORMATION_SCHEMA` references (since the commit-specific dataset may get created), but wouldn't work for `{project}.{region}.INFORMATION_SCHEMA` references (because something like `region-us_{hash}` isn't a valid region).

It seems like we should either not update `INFORMATION_SCHEMA` references at all (in which case there might be CI permissions issues depending on the project/dataset component), or have an additional check so that if it's a region qualifier we don't try to append the commit hash.

The current [CI problem](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/24247/workflows/f57818b1-52a6-4f04-8b1e-a9c1c6a48fcf/jobs/249913) for #4223 is with updating the `INFORMATION_SCHEMA` references causing invalid SQL, not with trying to deploy `INFORMATION_SCHEMA` artifacts to staging.

Reply:
Ah right...
Not updating the references will indeed create permission problems on deploy. So we'll need to check for the region qualifier

Reply:
Also, there might be an issue with quoting updated references, where if the `INFORMATION_SCHEMA.{view}` part is quoted separately it won't be valid.  If we don't want to special case that, we could avoid triggering that quoting style by fully quoting the original `INFORMATION_SCHEMA` references (e.g. `monitoring_derived.telemetry_missing_columns_v2` and `monitoring_derived.telemetry_missing_columns_v3` would need to be updated in that case).

Reply:
I updated the `monitoring_derived` tables and added some logic to prevent the stage deploy from adding the suffix to anything called `INFORMATION_SCHEMA` or `region-us`